### PR TITLE
perf(*): Speeding up the build, fixing transient test errors

### PR DESCRIPTION
**Pull request by robzienert**
I wanted to use Orca as a test bed for squeezing out some more build perf. We'll be able to get some more help from Gradle 4's scan capability, but until then, this is a small win.

# Gradle setting changes

* `./gradlew clean check`: Total time: 2 mins 20.529 secs
* `./gradlew clean check --parallel`: Total time: 1 mins 45.221 secs

I also tried `test.maxParllelForks=1+n`, but had performance regressions combined with `parallel`. Also tried `options.fork` on tasks with `JavaCompile`, but didn't see any noticeable difference.

Ran the tests multiple times on my laptop, and the numbers were all +/-5 seconds.

Also noticed we're not correctly cleaning up EmbeddedRedis when tests fail. Discovered this by encountering a max open files error on my laptop. This is easy enough to clean up `ps -ef | awk '/redis-server/{print $2}' | xargs kill -9`, but will want to look at this again soon.

# Removal of orca-loadtest

We don't use it. The integration-scoped approach from Adam's [spinnaker-performance](https://github.com/ajordens/spinnaker-performance) has proven easier and more effective.

# Transient test error

    Failures (2):
      JUnit Vintage:com.netflix.spinnaker.orca.q.redis.RedisQueueIntegrationTest:can run a stage with an execution window
        MethodSource [className = 'com.netflix.spinnaker.orca.q.QueueIntegrationTest', methodName = 'can run a stage with an execution window', methodParameterTypes = '']
        => org.junit.ComparisonFailure: expected:<[SUCCEEDED]> but was:<[TERMINAL]>
      JUnit Vintage:com.netflix.spinnaker.orca.q.redis.RedisQueueIntegrationTest:parallel stages do not duplicate execution windows
        MethodSource [className = 'com.netflix.spinnaker.orca.q.QueueIntegrationTest', methodName = 'parallel stages do not duplicate execution windows', methodParameterTypes = '']
        => org.junit.ComparisonFailure: expected:<[SUCCEEDED]> but was:<[TERMINAL]>

Fixed. Was a datetime math bug in the test code.

Was easy to track down via:

```
-      assertThat(status).isEqualTo(SUCCEEDED)
+      assertThat(status).withFailMessage("status is not succeeded:\n ${jacksonObjectMapper().enable(SerializationFeature.INDENT_OUTPUT).writeValueAsString(this)}").isEqualTo(SUCCEEDED)
```

### CODE REVIEW

**Comment by robfletcher** [comment]
Failed on Travis. Oh the irony.

File: orca-queue-tck/src/main/kotlin/com/netflix/spinnaker/orca/q/QueueIntegrationTest.kt [hunk 1]
```diff
@@ -429,9 +432,9 @@ class QueueIntegrationTest {
           "restrictedExecutionWindow" to mapOf(
             "days" to (1..7).toList(),
             "whitelist" to listOf(mapOf(
-              "startHour" to now().atZone(timeZone).hour,
+              "startHour" to nowHour,
               "startMin" to 0,
-              "endHour" to now().atZone(timeZone).hour + 1,
+              "endHour" to if (nowHour + 1 == 24) 0 else nowHour + 1,
```
Comment (robfletcher) [inline orca-queue-tck/src/main/kotlin/com/netflix/spinnaker/orca/q/QueueIntegrationTest.kt]: Oh, that was stupid of me.

Comment (robfletcher) [reply orca-queue-tck/src/main/kotlin/com/netflix/spinnaker/orca/q/QueueIntegrationTest.kt]: `now().atZone(timeZone).plus(HOURS, 1).hour` would have avoided the bug.

File: orca-queue-tck/src/main/kotlin/com/netflix/spinnaker/orca/q/QueueIntegrationTest.kt [hunk 2]
```diff
@@ -446,7 +449,7 @@ class QueueIntegrationTest {
     context.runToCompletion(pipeline, runner::start, repository)
 
     repository.retrieve(PIPELINE, pipeline.id).apply {
-      assertThat(status).isEqualTo(SUCCEEDED)
+      assertThat(status)isEqualTo(SUCCEEDED)
```
Comment (robfletcher) [inline orca-queue-tck/src/main/kotlin/com/netflix/spinnaker/orca/q/QueueIntegrationTest.kt]: Huh?

Comment (robfletcher) [reply orca-queue-tck/src/main/kotlin/com/netflix/spinnaker/orca/q/QueueIntegrationTest.kt]: @robzienert That's what's making it fail on Travis.

**Comment by robzienert** [comment]
Lol oops. Thanks for the typo fix

**Pull request merged**